### PR TITLE
chore: update eslint to v9

### PR DESCRIPTION
**Pull request by UziTech**
## Description

Update eslint to v9

This does mean removing standard since it is [no longer maintained](https://github.com/standard/standard/issues/1948#issuecomment-2138078249)

We could use [neostandard](https://github.com/neostandard/neostandard) which is the successor but I don't see why we wouldn't think the same thing will happen to it in the future. Just copying the config seems to be the best option to continue with the same rules.

Maybe a future PR can slim down the rules a bit. Personally I prefer only using `eslint:recommended` and `typescript-eslint:recommended` since it is simpler.

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

File: test/specs/redos/cubic_def.cjs [hunk 1]
```diff
@@ -1,4 +1,4 @@
 module.exports = {
   markdown: `[x]:${' '.repeat(1500)}x ${' '.repeat(1500)} x`,
-  html: `<p>[x]:${' '.repeat(1500)}x ${' '.repeat(1500)} x</p>`,
+  html: `<p>[x]:${' '.repeat(1500)}x ${' '.repeat(1500)} x</p>`
```
Comment (styfle) [inline test/specs/redos/cubic_def.cjs:3]: Personally, I like trailing commas

File: package.json [hunk 2]
```diff
@@ -83,30 +79,32 @@
...
-    "test:update": "node test/update-specs.js",
-    "rules": "node test/rules.js",
-    "bench": "npm run build && node test/bench.js",
-    "lint": "eslint --fix .",
-    "build:reset": "git checkout upstream/master lib/marked.cjs lib/marked.umd.js lib/marked.esm.js marked.min.js",
-    "build": "npm run rollup && npm run build:types && npm run build:man",
-    "build:docs": "npm run build && node docs/build.js",
-    "build:types": "tsc && dts-bundle-generator --project tsconfig.json -o lib/marked.d.ts src/marked.ts && dts-bundle-generator --project tsconfig.json -o lib/marked.d.cts src/marked.ts",
-    "build:man": "marked-man man/marked.1.md > man/marked.1",
-    "rollup": "rollup -c rollup.config.js"
```
Comment (styfle) [inline package.json:109]: Whats changing in the scripts here? Is it just the order?

**Review by styfle — APPROVED** [review]

File: test/specs/redos/cubic_def.cjs (hunk 1 above)
Comment (UziTech) [reply test/specs/redos/cubic_def.cjs:3]: Me too. I was just kept the rules the same as before.

I will try changing the rules in a follow up pr. There will be a lot of src code changes

File: package.json (hunk 2 above)
Comment (UziTech) [reply package.json:109]: Ya just reordering alphabetically

**Pull request merged**